x-pack/legacy/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -350,6 +371,45 @@ export class VectorLayer extends AbstractLayer {
     };
   }
 
+  async _performInnerJoins(sourceResult, joinStates, updateSourceData) {
+
+    //should update the store iff
+    //-- source result was refreshed
+    //-- any of the join configurations changed (joinState changed)
+    //-- visibility of any of the features has changed
+
+    let shouldUpdateStore = sourceResult.refreshed || joinStates.some((joinState) => joinState.dataHasChanged);
+
+    if (!shouldUpdateStore) {
+      return;
+    }
+
+    for (let i = 0; i < sourceResult.featureCollection.features.length; i++) {
+      const feature = sourceResult.featureCollection.features[i];
+      const oldVisbility = feature.properties[FEATURE_VISIBLE_PROPERTY_NAME];
+
+
+      let canDoAllJoins = true;
```

Comment:
How about renaming `canDoAllJoins` to  `isFeatureVisible`? That way, when the variable is used, the name alines with the use instead of the condition you are looking for

```
if (oldVisbility !== isFeatureVisible) {
  shouldUpdateStore = true;
}

feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = isFeatureVisible;
```